### PR TITLE
Added timeout block in repository-function module.

### PR DESCRIPTION
**Pull request by fbeevikm**
Issue : https://github.com/terraform-google-modules/terraform-google-event-function/issues/98

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@fbeevikm
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/repository-function/main.tf [hunk 1]
```diff
@@ -49,4 +49,14 @@ resource "google_cloudfunctions_function" "main" {
   project               = var.project_id
   region                = var.region
   service_account_email = var.service_account_email
+
+
+  dynamic "timeouts" {
+    for_each = var.operation_timeouts
+    content {
+      create = timeouts.value["create"]
```
Comment (bharathkkb) [inline modules/repository-function/main.tf:57]: We may have to a lookup here for default value.

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @fbeevikm 

Here is the error from CI
```
This value does not have any indices.
╵}
    apply.go:34: 
        	Error Trace:	apply.go:34
        	            				terraform.go:303
        	            				terraform.go:313
        	            				terraform.go:336
        	            				stages.go:31
        	            				terraform.go:336
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; ╷
        	            	│ Error: Invalid index
        	            	│ 
        	            	│   on ../../modules/repository-function/main.tf line 57, in resource "google_cloudfunctions_function" "main":
        	            	│   57:       create = timeouts.value["create"]
        	            	│     ├────────────────
        	            	│     │ timeouts.value is "10m"
        	            	│ 
        	            	│ This value does not have any indices.
        	            	╵
        	            	╷
        	            	│ Error: Invalid index
        	            	│ 
        	            	│   on ../../modules/repository-function/main.tf line 58, in resource "google_cloudfunctions_function" "main":
        	            	│   58:       update = timeouts.value["update"]
        	            	│     ├────────────────
        	            	│     │ timeouts.value is "10m"
        	            	│ 
        	            	│ This value does not have any indices.
        	            	╵
        	            	╷
        	            	│ Error: Invalid index
        	            	│ 
        	            	│   on ../../modules/repository-function/main.tf line 59, in resource "google_cloudfunctions_function" "main":
        	            	│   59:       delete = timeouts.value["delete"]
        	            	│     ├────────────────
        	            	│     │ timeouts.value is "10m"
        	            	│ 
        	            	│ This value does not have any indices.
        	            	╵}
        	Test:       	TestAll/examples/automatic-labelling-from-repository
```

File: modules/repository-function/main.tf (hunk 1 above)
Comment (fbeevikm) [reply modules/repository-function/main.tf]: Added optional in variable. it's an experimental feature. Please let me know if you any other suggestions.

**Comment by comment-bot-dev** [comment]
@fbeevikm
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
Checking for missing newline at end of file
Error: No newline at end of file ./modules/repository-function/main.tf
Error: No newline at end of file ./modules/repository-function/variables.tf
 ```

**Comment by comment-bot-dev** [comment]
@fbeevikm
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@fbeevikm
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@fbeevikm
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**